src/main/scala/org/opennetworkinsight/dns/DNSPostLDA.scala

```diff
@@ -142,8 +143,28 @@ object DNSPostLDA {
     addcol("score")
 
     logger.info("Persisting data")
-    val scored = src_scored.filter(elem => elem._1 < threshold).sortByKey().map(row => row._2.mkString(","))
-    scored.saveAsTextFile(resultsFilePath)
+
+
+    val filtered = src_scored.filter(elem => elem._1 < threshold)
+
+    val count = filtered.count
+
+    val takeCount  = if (topK == -1 || count < topK) {
+      count.toInt
+    } else {
+      topK
+    }
+    class OrderByLeastScoreToTop() extends Ordering[(Double,Array[Any])] {
+      def compare(p1: (Double, Array[Any]), p2: (Double, Array[Any]))    = p2._1.compare(p1._1)
+    }
+
+    implicit val ordering = new OrderByLeastScoreToTop()
+
+    val top : Array[(Double,Array[Any])] = filtered.top(takeCount)
```

Comment:
that's why OrderedByLeastScoreToTop exists... so that top will return the least scored elements

notice how it takes the reverse of the standard numerical ordering:
def compare(row1: Row, row2: Row) = row2.getDouble(scoreIndex).compare(row1.getDouble(scoreIndex))

**_BUT...._**

it would be simpler to use the standard numerical ordering and use the takeOrdered function